lib/apiservers/engine/backends/volume.go

```diff
@@ -58,7 +105,43 @@ func (v *Volume) VolumeRm(name string) error {
 		if _, ok := err.(*storage.RemoveVolumeConflict); ok {
 			return derr.NewRequestConflictError(fmt.Errorf("Volume is in use"))
 		}
-		return derr.NewErrorWithStatusCode(fmt.Errorf("Server error form portlayer"), http.StatusInternalServerError)
+		return derr.NewErrorWithStatusCode(fmt.Errorf("Server error from portlayer: %s", err), http.StatusInternalServerError)
 	}
 	return nil
 }
+
+//Utility Functions
+
+func fillDockerVolumeModel(volume *models.VolumeResponse, labels map[string]string) *types.Volume {
+	dockerVol := types.Volume{
+		Driver:     volume.Driver,
+		Name:       volume.Name,
+		Labels:     labels,
+		Mountpoint: volume.SelfLink,
+	}
+	return &dockerVol
+}
+
+func translateInputsToPortlayerRequestModel(name, host, driverName string, opts, labels map[string]string) models.VolumeRequest {
+	model := models.VolumeRequest{
+		Driver:     driverName,
+		DriverArgs: opts,
+		Name:       name,
+		Store:      host,
```

Comment:
To extend our example, given the following docker user input

> docker volume create --driver=vsphere --name="nameRelevantToMeTheDockerUser" -o "volumestore=Fast" capacity=8G foo=bar something=somethingelse

The call to the portlayer should be 

`VolumeCreate(ctx, "nameRelelvantToMeTheDockerUser", util.VolumeStore("Fast"), 8 * 1024, map[string][]byte{"foo": []byte("bar"), "something":[]byte("somethingelse")})`